site/_docs/avatica_json_reference.md

```diff
@@ -0,0 +1,666 @@
+
+`statementId` (required integer) The identifier of the statement created using the above connection.
+
+`offset` (required integer) The positional offset into a result set to fetch.
+
+`fetchMatchRowCount` (required integer) The maximum number of rows to return in the response to this request.
+
+`parameterValues` (optional array of nested objects) The types of the object to set on the prepared statement in use.
+
+### PrepareAndExecuteRequest
+
+{% highlight json %}
+{
+  connectionId: "000000-0000-0000-00000000",
+  statementId: 12345,
+  sql: "SELECT * FROM table",
+  maxRowCount: 100,
+}
+{% endhighlight %}
+
+`connectionId` (required string) The identifier for the connection to use.
+
+`statementId` (required integer) The identifier for the statement created by the above connection to use.
+
+`sql` (required string) A SQL statement
+
+`maxRowCount` (required long) The maximum number of rows returned in the response.
+
+### PrepareRequest
+
+{% highlight json %}
+{
+  connectionId: "000000-0000-0000-00000000",
+  sql: "SELECT * FROM table",
+  maxRowCount: 100,
+}
+{% endhighlight %}
+
+`connectionId` (required string) The identifier for the connection to use.
+
+`sql` (required string) A SQL statement
+
+`maxRowCount` (required long) The maximum number of rows returned in the response.
+
+### SchemasRequest
+
+{% highlight json %}
+{
+  catalog: "name",
+  schemaPattern: "pattern.*"
+}
+{% endhighlight %}
+
+`catalog` (required string) The name of the catalog to fetch the schema from.
+
+`schemaPattern` (required string) A Java pattern of schemas to fetch.
+
+### TableTypesRequest
+
+{% highlight json %}
+{}
+{% endhighlight %}
+
+This request has no attributes.
+
+### TablesRequest
+
+{% highlight json %}
+{
+  catalog: "catalog_name",
+  schemaPattern: "schema_pattern.*",
+  tableNamePattern: "table_name_pattern.*",
+  typeList: [ "TABLE", "VIEW", ... ]
+}
+{% endhighlight %}
+
+`catalog` (optional string) The name of a catalog to restrict fetched tables.
+
+`schemaPattern` (optional string) A Java Pattern representing schemas to include in fetched tables.
+
+`tableNamePattern` (optional string) A Java Pattern representing table names to include in fetched tables.
+
+`typeList` (optional array of string) A list of table types used to restrict fetched tables.
+
+### TypeInfoRequest
+
+{% highlight json %}
+{}
+{% endhighlight %}
+
+This request has no attributes.
+
+## Responses
+
+The collection of all JSON objects returned as responses from Avatica.
+
+### ResultSetResponse
+
+{% highlight json %}
+{
+  connectionId: "000000-0000-0000-00000000",
+  statementId: 12345,
+  ownStatement: true,
+  signature: Signature,
+  firstFrame: Frame,
+  updateCount: 10
+}
+{% endhighlight %}
+
+`connectionId` The identifier for the connection used to generate this response.
+
+`statementId` The identifier for the statement used to generate this response.
+
+`ownStatement` TODO
```

Comment:
"Whether the result set has its own dedicated statement. If true, the server must automatically close the statement when the result set is closed. This is used for JDBC metadata result sets, for instance."